### ISSUE TITLE
Bug: VBA edit - Attribute error

### ISSUE THREAD

**Issue by Cralost** [xlwings/xlwings#2148]
When I edit code in VS code and save, everything works fine except importing attributes. 

I have keyboard shortcut assigned to macro (Ctrl+L). When I edit code in VS code I can see

`Attribute Module1.VB_ProcData.VB_Invoke_Func = "l\n14"` it the function - which is totally fine. 

However when I save and xlwings imports it into workbook - it imports it as whole text, what VBE cannot handle then:

![image](https://user-images.githubusercontent.com/77323021/213871869-3deeb4f5-6c44-434a-a055-ded0d4509305.png)

But when I import the same .bas file via VBE interface `Modules -> Right Click -> Import Modules...` everything is fine. 

I already saw this behavior when I was pasting entire text.  

I can see in the code that it does some kind of import, but have not found what that Import function actually does: 

https://github.com/xlwings/xlwings/blob/303dc76edff10088f23fe9e9e4ebb8878767b7c7/xlwings/cli.py#L716

Does it mimic the native VBE Import function or it imports modules as text?

**Comment by fzumstein** [comment]
This looks like a similar issue to https://github.com/xlwings/xlwings/issues/2088: the current logic with regards to these special commands isn't good enough (I don't need it for the puprose of xlwings). You're linking to the wrong function, however. The correct one would be: https://github.com/xlwings/xlwings/blob/303dc76edff10088f23fe9e9e4ebb8878767b7c7/xlwings/cli.py#L875

**Comment by AtheusAlves** [comment]
Boa noite, pessoal. 
Minha dúvida tá bem longe do problema acima, peço desculpas por usar esse espaço para isso.

Estou tentando criar uma função, onde coloco " , " depois de cada número ou palavra.
Segue o script do vba:

Public Function Virgula(Texto1, Texto2)
Virgula = Texto1 & "," & Texto2 & ","
    
End Function

Só consigo até dois, mas queria até 1000, se possível.
É algo que irá me ajudar nos Scripts do sql.

**Comment by Cralost** [comment]
Yeah, but does it actually work with official VBA Import function or it just try to replicate that function by replacing plain text? Because if only as plain text, then approach with just removing attributes would cause removing some "properties" as in my case keyboard shortcut. 

Do you think this could be done or is it just limitation of vba_edit and all I have to do is to get away with it?

**Comment by fzumstein** [comment]
`vba edit` replaces the text directly in the VBA module, but `vba import` removes the module and re-imports the source file which would honour your keyboard shortcut.

**Comment by Cralost** [comment]
I was looking around and now I wonder, why not to use `book.api.VBProject.VBComponents.Import` right away instead of `vb_component.CodeModule.AddFromString` ?

**Comment by fzumstein** [comment]
I first did use it (https://github.com/xlwings/xlwings/commit/2799bf1c77dd05f7f576ecad35fd09604e8ec78d) but since ThisWorkbook/Sheet would require AddFromString, it was simplifying everything and it's also faster. But yeah, it's not a big deal for me, maybe it should be changed back.
